### PR TITLE
add bucket replicate component

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <ben.ye@bytedance.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Add a new bucket replicate component. The libsonnet file is modified based on the bucket.libsonnet file.

## Verification

We have been using this component in production.

### CODE REVIEW

**Comment by yeya24** [comment]
cc @kakkoyun @metalmatze

**Review by kakkoyun — APPROVED** [review]
LGTM. I don't get why minio related manifests have changed. Did we miss something feom previous PRs?

File: examples/development-minio/pvc.yaml [hunk 1]
```diff
@@ -2,10 +2,10 @@ apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
   name: minio
-  namespace: thanos
```
Comment (yeya24) [inline examples/development-minio/pvc.yaml:5]: I think we can remove this line as other minio manifests don't have namespace specified as well.

**Comment by yeya24** [comment]
> LGTM. I don't get why minio related manifests have changed. Did we miss something feom previous PRs?

Thanks for catching this. I updated some minio manifests mistakenly and I already changed them back. PTAL

**Comment by yeya24** [comment]
Ready for review again. CC @kakkoyun

**Pull request merged**